### DISCUSSION TITLE
Get PR repository name

### DISCUSSION THREAD

**Discussion by reikdas** [community/community#26411] [Actions]
<p>Here - <a href="https://github.com/root-project/cling/actions/runs/341307211/workflow#L113" rel="noopener nofollow ugc">https://github.com/root-project/cling/actions/runs/341307211/workflow#L113</a> I use the <code>GITHUB_REPOSITORY</code> environment variable to get the name of the repository that a PR is made to or a commit is pushed. Instead, I want to get the name of the repository that the PR is coming from. If it is a fork, then the name of the fork and if it is from a branch of the main repository, then the name of the main repository.<br>
Is an environment variable available that would give me this information?</p>

**Comment by Simran-B** [comment, marked as answer]
<p>You can do e.g.</p>
<pre><code class="lang-yaml">- env:
    REPO: ${{ github.event.pull_request.head.repo.full_name }}
  run: |
    echo "$REPO"
  if: github.event_name == 'pull_request'
</code></pre>
<p>REPO will be like <code>other-user/repo-name</code>.</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/developers/webhooks-and-events/webhook-events-and-payloads#webhook-payload-example-30" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="" height="">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/developers/webhooks-and-events/webhook-events-and-payloads#webhook-payload-example-30" target="_blank" rel="noopener">Webhook events and payloads - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by reikdas** [comment]
<p>Thanks! I would ideally like to be able to use the same variable for both <code>$GITHUB_REPOSITORY</code> and <code>github.event.pull_request.head.repo.full_name</code> if possible.</p>
<p>Is something like -</p>
<pre><code class="lang-auto">- env:
    if [ ${{ github.event_name }} == 'pull_request' ]; then
        REPO: ${{ github.event.pull_request.head.repo.full_name }}
    else
        REPO: $GITHUB_REPOSITORY
   run: |
        xyz
        abcd
   shell: bash
</code></pre>
<p>possible?</p>

**Comment by airtower-luna** [comment]
<p>Almost, you’d have to either write <code>REPO</code> to <code>$GITHUB_ENV</code> in one step and use it in another, or for just one step set <code>REPO</code> right within the <code>run</code> part.</p>
<p>Version with multiple steps:</p>
<pre><code class="lang-yaml">- run: |
    if [ ${{ github.event_name }} = 'pull_request' ]; then
        echo "REPO=${{ github.event.pull_request.head.repo.full_name }}" &gt;&gt;$GITHUB_ENV
    else
        echo "REPO=$GITHUB_REPOSITORY" &gt;&gt;$GITHUB_ENV
    fi
- run: echo "REPO is ${REPO}"
</code></pre>
<p>Version with one step (<code>REPO</code> is available only within that step!):</p>
<pre><code class="lang-yaml">- run: |
    if [ ${{ github.event_name }} = 'pull_request' ]; then
        REPO="${{ github.event.pull_request.head.repo.full_name }}"
    else
        REPO="$GITHUB_REPOSITORY"
    fi
    echo "REPO is ${REPO}"
</code></pre>

**Comment by Simran-B** [comment]
<p>Seems unnecessary complicated, I would try this:</p>
<pre><code>REPO="${{ github.event.pull_request.head.repo.full_name || github.repository }}"</code></pre>

**Comment by reikdas** [comment]
<aside class="quote no-group" data-username="airtower-luna" data-post="4" data-topic="141133">
<div class="title">
<div class="quote-controls"></div>
 airtower-luna:</div>
<blockquote>
<p>Version with one step ( <code>REPO</code> is available only within that step!):</p>
<pre><code class="lang-auto">- run: |
    if [ ${{ github.event_name }} = 'pull_request' ]; then
        REPO="${{ github.event.pull_request.head.repo.full_name }}"
    else
        REPO="$GITHUB_REPOSITORY"
    fi
    echo "REPO is ${REPO}"
</code></pre>
</blockquote>
</aside>
<aside class="quote no-group" data-username="Simran-B" data-post="5" data-topic="141133" data-full="true">
<div class="title">
<div class="quote-controls"></div>
 Simran-B:</div>
<blockquote>
<p>Seems unnecessary complicated, I would try this:</p>
<pre><code class="lang-auto">REPO="${{ github.event.pull_request.head.repo.full_name || github.repository }}"
</code></pre>
</blockquote>
</aside>
<p>I have a followup question - how would I do this in Windows cmd?</p>
<p><code>IF (${{ github.event_name }}=="pull_request") (set REPO="${{ github.event.pull_request.head.repo.full_name }}") else (set REPO="%GITHUB_REPOSITORY%")</code></p>
<p>doesn’t seem to work.</p>

**Comment by Simran-B** [comment]
<pre><code class="lang-auto">set REPO=${{ github.event.pull_request.head.repo.full_name || github.repository }}
echo %REPO%
</code></pre>
<p>or</p>
<pre><code class="lang-auto">IF "${{ github.event_name }}"=="pull_request" (
  set REPO=${{ github.event.pull_request.head.repo.full_name }}
) ELSE (
  set REPO=%GITHUB_REPOSITORY%
)
echo %REPO%
</code></pre>
<p>There are apparently a few things to watch out for in batch scripts:</p>
<ul>
<li>Don’t wrap the if condition in parentheses. This seems to make the condition fail!</li>
<li>Use double quotes on both sides of a comparison for equality:
<ul>
<li>
<code>IF a==a (echo a) ELSE (echo b)</code> → a</li>
<li>
<code>IF a=="a" (echo a) ELSE (echo b)</code> → b</li>
<li>
<code>IF "a"=="a" (echo a) ELSE (echo b)</code> → a</li>
</ul>
</li>
<li>Don’t use quote marks for <code>set</code> unless you actually want them in the string. There is no need to wrap strings in quotes that have spaces in them in case of <code>set</code>.</li>
<li>Place a space between <code>ELSE</code> and <code>(</code> or it’s a syntax error</li>
</ul>

**Comment by Simran-B** [comment]
<p>The log says:</p>
<blockquote>
<p>shell: C:\Program Files\PowerShell\7\pwsh.EXE -command “. ‘{0}’”</p>
</blockquote>
<p>Looks like you’re simply using the wrong shell type. The default for Windows runners is <code>pwsh</code>. You should explicitly set <code>shell: cmd</code> if you want to run batch code.</p>
<p>The same in PowerShell would be:</p>
<pre><code class="lang-powershell">if ('${{ github.event_name }}' -eq 'pull_request') {
  $REPO = '${{ github.event.pull_request.head.repo.full_name }}'
} else {
  $REPO = $Env:GITHUB_REPOSITORY
}
echo $REPO
</code></pre>
